homeassistant/components/mqtt/__init__.py

```diff
@@ -370,18 +389,30 @@ def async_publish_service(call):
         msg_topic = call.data[ATTR_TOPIC]
         payload = call.data.get(ATTR_PAYLOAD)
         payload_template = call.data.get(ATTR_PAYLOAD_TEMPLATE)
+        payload_file_path = call.data.get(ATTR_PAYLOAD_FILE_PATH)
         qos = call.data[ATTR_QOS]
         retain = call.data[ATTR_RETAIN]
-        try:
-            if payload_template is not None:
+        if payload_template is not None:
+            try:
                 payload = \
                     template.Template(payload_template, hass).async_render()
-        except template.jinja2.TemplateError as exc:
-            _LOGGER.error(
-                "Unable to publish to '%s': rendering payload template of "
-                "'%s' failed because %s",
-                msg_topic, payload_template, exc)
-            return
+            except template.jinja2.TemplateError as exc:
+                _LOGGER.error(
+                    "Unable to publish to '%s': rendering payload template of "
+                    "'%s' failed because %s",
+                    msg_topic, payload_template, exc)
+                return
+        elif payload_file_path is not None:
```

Comment:
I'm deeply sorry @balloob and @pvizeli to come back here, just tell me if I have to move this question somewhere else more appropriate.

So, I'm working on a notifier component that sends images through MQTT. `mqtt.async_publish` works fine and the image is sent and received properly.

I have two problems: the first one is that the image data gets INFO logged, I guess by [this](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/core.py#L388), obviously spamming the log.

The second one is that the Recorder throws an error:
`ERROR (Recorder) [homeassistant.components.recorder.util] Error executing query: bytearray(THE WHOLE PICTURE CONTENT, AGAIN) is not JSON serializable`

I have no idea how to solve this, it's too hard 'core' ;)

Could you please suggest me how to move from here?

Thanks!
gl